images/airflow/2.9.2/python/mwaa/entrypoint.py

```diff
@@ -190,15 +190,31 @@ def create_queue() -> None:
 
 
 def _requirements_has_constraints(requirements_file: str):
-    with open(requirements_file, "r") as file:
-        for line in file:
-            # Notice that this regex check will also match lines with commented out
-            # constraints flag. This is intentional as a mechanism for users who
-            # want to avoid enforcing the default Airflow constraints, yet don't want
-            # to provide a constraints file.
-            if re.search(r"-c |--constraint ", line):
-                return True
+    encodings_to_try = ["utf-8", "ascii", "latin1"]
+    for encoding in encodings_to_try:
+        try:
+            with open(requirements_file, "r", encoding=encoding) as file:
+                for line in file:
+                    # Notice that this regex check will also match lines with commented
+                    # out constraints flag. This is intentional as a mechanism for users
+                    # who want to avoid enforcing the default Airflow constraints, yet
+                    # don't want to provide a constraints file.
+                    if re.search(r"-c |--constraint ", line):
+                        return True
+            return False
```

Comment:
If first encoding is not successful, will it not end the loop per-maturely ?

Reply:
No. If the reading fails due to encoding, it will cause an exception, and the next encoding will be tried. If the reading was successful, and we couldn't find the `-c/--constraint` flag, then we return False.